File: app/src/main/java/jgappsandgames/smartreminderslite/status/StatusActivityInterface.java
package jgappsandgames.smartreminderslite.status;

import android.app.Activity;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import jgappsandgames.smartreminderslite.R;
import jgappsandgames.smartreminderslite.holder.TaskFolderHolder;

/**
 * StatusActivityInterface
 * Created by joshua on 11/29/2017.
 */
public abstract class StatusActivityInterface extends Activity implements TaskFolderHolder.OnTaskChangedListener {
    // Views
    protected TextView overdue_text;
    protected ListView overdue_list;
    protected TextView incomplete_text;
    protected ListView incomplete_list;
    protected TextView done_text;
    protected ListView done_list;

    // Adapters
    protected StatusAdapter overdue_adapter;
    protected StatusAdapter incomplete_adapter;
    protected StatusAdapter done_adapter;

    // LifeCycle Methods
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Set Content View
        setContentView(R.layout.activity_status);

        // Find Views
        overdue_text = findViewById(R.id.overdue_text);
        overdue_list = findViewById(R.id.overdue_list);
        incomplete_text = findViewById(R.id.incomplete_text);
        incomplete_list = findViewById(R.id.incomplete_list);
        done_text = findViewById(R.id.done_text);
        done_list = findViewById(R.id.done_list);
    }

    // Menu
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_auxilary, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.save:
                save();
                Toast.makeText(this, "Saved.", Toast.LENGTH_LONG).show();
                break;

            case R.id.close:
                finish();
        }

        return super.onOptionsItemSelected(item);
    }

    // Abstract Methods
    public abstract void save();
}

File: app/src/main/java/jgappsandgames/smartreminderslite/sort/status/StatusActivityInterface.kt
package jgappsandgames.smartreminderslite.sort.status

// Android OS
import android.app.Activity
import android.os.Bundle

// Views
import android.view.Menu
import android.view.MenuItem
import android.widget.ListView
import android.widget.TextView

// Anko
import org.jetbrains.anko.toast

// App
import jgappsandgames.smartreminderslite.R
import jgappsandgames.smartreminderslite.holder.TaskFolderHolder

/**
 * StatusActivityInterface
 * Created by joshua on 12/14/2017.
 *
 * Status View[StatusActivityInterface, StatusActivity]
 * A View for Viewing Tasks Based on Their Status
 *
 *     StatusActivityInterface: View and Application Focus
 *     StatusActivity: Data and User Input
 */
abstract class StatusActivityInterface: Activity(), TaskFolderHolder.OnTaskChangedListener {
    // Views ---------------------------------------------------------------------------------------
    protected var overdueText: TextView? = null
    protected var overdueList: ListView? = null
    protected var incompleteText: TextView? = null
    protected var incompleteList: ListView? = null
    protected var completeText: TextView? = null
    protected var completeList: ListView? = null

    // LifeCycle Methods ---------------------------------------------------------------------------
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_status)

        // Find Views
        overdueText = findViewById(R.id.overdue_text)
        overdueList = findViewById(R.id.overdue_list)
        incompleteText = findViewById(R.id.incomplete_text)
        incompleteList = findViewById(R.id.incomplete_list)
        completeText = findViewById(R.id.done_text)
        completeList = findViewById(R.id.done_list)
    }

    // Menu Methods --------------------------------------------------------------------------------
    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_auxilary, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item!!.itemId) {
            R.id.save -> {
                save()
                toast(R.string.saved).show()
                return true
            }

            R.id.close -> {
                finish()
                return true
            }
        }

        return false
    }

    // Abstract Methods ----------------------------------------------------------------------------
    protected abstract fun save()
}